hazelcast/src/main/java/com/hazelcast/internal/blackbox/impl/BlackboxImpl.java

```diff
@@ -0,0 +1,315 @@
+     * are not going to fail. The blackbox should never fail unless the arguments don't make any sense of course.
+     *
+     * @param parameter
+     * @param sensor
+     */
+    private void checkOverwrite(String parameter, SensorImpl sensor) {
+        // if an input already exists, we are just going to overwrite it.
+        if (sensor.input != null) {
+            logger.warning(format("Duplicate registration, a SensorInput for parameter '%s' already exists", parameter));
+        }
+
+        // if a source already exists, we are just going to overwrite it.
+        if (sensor.source != null) {
+            logger.warning(format("Duplicate registration, a source for parameter '%s' already exists", parameter));
+        }
+    }
+
+    @Override
+    public SensorView getView(String parameter) {
+        checkNotNull(parameter, "parameter can't be null");
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.get(parameter);
+
+            if (sensor == null) {
+                sensor = new SensorImpl(parameter, this);
+                sensors.put(parameter, sensor);
+            }
+
+            if (sensor.view != null) {
+                return sensor.view;
+            }
+
+            SensorViewImpl view = new SensorViewImpl(this, parameter, logger);
+            sensor.view = view;
+            view.input = sensor.input;
+            return view;
+        }
+    }
+
+    public void destroy(String parameter) {
+        checkNotNull(parameter, "parameter can't be null");
+
+        synchronized (getLock(parameter)) {
+            SensorImpl sensor = sensors.remove(parameter);
+            if (sensor == null) {
+                return;
+            }
+
+            if (sensor.view != null) {
+                sensor.view.input = null;
+            }
+            sensor.input = null;
+            sensor.view = null;
+
+            if (logger.isFinestEnabled()) {
+                logger.finest("Destroying sensor " + parameter);
+            }
+
+            modCount.incrementAndGet();
+        }
+    }
+
+    @Override
+    public void scheduleAtFixedRate(Runnable publisher, long delay, TimeUnit timeUnit) {
+        scheduledExecutorService.scheduleAtFixedRate(publisher, 0, delay, timeUnit);
+    }
+
+    boolean containsSensor(String parameter) {
+        return sensors.containsKey(parameter);
+    }
+
+    @Override
+    public Sensor register(String parameter, final AtomicLong atomicLong) {
+        checkNotNull(atomicLong, "atomicLong can't be null");
+
+        LongSensorInput input = new LongSensorInput() {
+            @Override
+            public long get() {
+                return atomicLong.get();
+            }
+        };
+
+        return register(parameter, input);
+    }
+
+    @Override
+    public Sensor registerLongField(String parameter, final Object object, String inpput) {
+        final Field field = getLongField(parameter, object, inpput);
```

Comment:
Would be a good addition for HZ 3.6. 

There are many improvements that can be made. E.g. also simplifying volatile writes to a lazySet. 

Main goal is to provide a way to look inside a HZ environment that is stable. Doesn't have much performance overhead and can help us out over the next 6 months.